### ISSUE TITLE
ProgressCircle: Fill color that indicates progress is not visible in HCM for all variants

### ISSUE THREAD

**Issue by dineshy87** [adobe/react-spectrum#4069]
# 🐛 Bug Report 

Fill color must be set to Highlight in high contrast mode. Currently, the fill color is not shown in High Contrast Mode. Currently, the fill color is changed in the variable but not set using border-color property yet.

## 🤔 Expected Behavior

'Highlight' color must be used as fill color. 

## 😯 Current Behavior

https://user-images.githubusercontent.com/16927765/219135762-2becaa4e-5c96-4d13-9e29-4075fa3195e4.mov

## 💁 Possible Solution

Setting border-color property with `!important` will apply Highlight as fill color for all variants. If !important is not preferred, it needs to be applied separately for indeterminate and overBackground variants using selectors for those variants.

**Change this**
``` CSS
@media (forced-colors: active) {
 .spectrum-CircleLoader-fill_7c1153 {
      --spectrum-loader-circle-medium-track-fill-color: Highlight;
 }
}

```

**To this**
``` CSS
@media (forced-colors: active) {
 .spectrum-CircleLoader-fill_7c1153 {	      
      --spectrum-loader-circle-medium-track-fill-color: Highlight;      
      border-color: var(--spectrum-loader-circle-medium-track-fill-color) !important;
 }
}

```

## 🔦 Context

Issue can be observed on [Storybook](https://reactspectrum.blob.core.windows.net/reactspectrum/41e26fa0f01ee9ab6e5c21edf6f7bb5d498bf40f/storybook/index.html?path=/story/progress-progresscircle--default&providerSwitcher-express=false&providerSwitcher-theme=lightest&providerSwitcher-toastPosition=bottom) and [react-spectrum docs](https://react-spectrum.adobe.com/react-spectrum/ProgressCircle.html) on all variants in HCM

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 
| Browser          | Edge 108|
| Operating System | Windows |

**Comment by majornista** [comment]
This may be fixed by: https://github.com/adobe/react-spectrum/pull/3827/files#diff-c6bd97135bfde61a860f3ced8fc7b020fbd8c46441d9f3b686717ffb6bf66b61.

**Comment by dineshy87** [comment]
Verified. Closing this. Thanks
